File: examples/python/wrappers/templates/stack_usage_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "cxx/templates/stack_usage.hpp"
#include "cxx/classes/project.hpp"

namespace py = pybind11;

void bindStackUsage(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::StackUsage> stackusage(m, "StackUsage");

    stackusage.def(py::init<>());
    stackusage.def("first_item_of_specialized", &iegen::example::StackUsage::firstItemOfSpecialized, py::arg("p"));
    stackusage.def("first_item_of_template", &iegen::example::StackUsage::firstItemOfTemplate<iegen::example::Project>, py::arg("arg0"));
}

File: examples/primitives/python/src/templates/stack_usage_pygen.py
from __future__ import annotations
from typing import *
import examples.templates.stack_usage as pybind_stack_usage_pygen
from bind_utils import *
from metaclass import *
from templates.stack_pygen import *
from classes.project_pygen import *


class StackUsage(metaclass=OriginalMethodsMetaclass):
    """
    comments
    """
    
    @bind
    def __init__(self):
        pass
    
    @bind
    def first_item_of_specialized(self, p: StackProject) -> Project:
        """
        comments
        """
        pass

    @bind
    def first_item_of_template(self, arg0: StackProject) -> Project:
        """
        comments
        """
        pass
